examples/anywhere_helper.rs

```diff
@@ -0,0 +1,133 @@
+//! This example shows an alternative way to implement parsing like
+//! the `anywhere_manual` example that involves defining a helper struct
+//! to contain the logic. (hey I heard you like state machines, so I...)
+//!
+//! Try it with arguments like: `x -r hi y -ohi z -- --not-a-flag`
+//!
+//! You will get:
+//!
+//! ```text
+//! positional: "x"
+//! option Short('r'): Ok("hi")
+//! positional: "y"
+//! option Short('o'): Some("hi")
+//! positional: "z"
+//! positional: "--not-a-flag"
+//! ```
+
+use crate::anywhere::{Opt, OptionsAnywhere};
+use std::env::args;
+
+#[allow(unused)] // This is reusable code; essentially a library module.
```

Comment:
Perhaps we should include this in getargs directly, instead of letting the user copy-paste this huge chunk of code.

Reply:
I would, but I can't find a good name for the enum, and I don't want to pollute `Opt` with the extra variant.